django_date_extensions/fields.py

```diff
@@ -39,26 +43,40 @@ def __init__(self, year=0, month=0, day=0, future=False, past=False):
         else:
             raise ValueError("You must specify a year")
 
+        # validate prefix
+        if prefix:
+            if not settings.ALLOWED_PREFIX:
+                raise ValueError("Prefix not allowed")
+            elif month or day:
+                raise ValueError("Prefix can only be set with Year only date")
+            elif not PREFIX_RE.match(prefix):
+                raise ValueError("Prefix value can only contains alphabets and can have maximum of 5 characters")
```

Comment:
This maximum isn't checked anywhere (though I don't think it should specify a maximum anyway).